EndlessClient/Dialogs/TextMultiInputDialog.cs

```diff
@@ -154,9 +201,68 @@ protected override void OnDrawControl(GameTime gameTime)
             _spriteBatch.Begin();
             _spriteBatch.Draw(_background, DrawAreaWithParentOffset, _backgroundSourceRectangle, Color.White);
             _spriteBatch.Draw(_background, DrawPositionWithParentOffset + _bottomOverlayDrawPosition, _bottomOverlaySource, Color.White);
+            _spriteBatch.Draw(_background, DrawPositionWithParentOffset + _rightOverlayDrawPosition, _rightOverlaySource, Color.White);
+
+            _spriteBatch.Draw(_background, DrawPositionWithParentOffset + _rightOverlayDrawPosition, _rightOverlaySource, Color.White);
+
             _spriteBatch.End();
 
             base.OnDrawControl(gameTime);
         }
+
+        // OnUpdateControl needs to solely run if Dialog.Nine is in use
+
+        protected override void OnUpdateControl(GameTime gameTime)
+        {
+            if (_dialogSize != DialogSize.NineWithScroll)
+            {
+                base.OnUpdateControl(gameTime);
+                return;
+            }
+            _scrollBar.UpdateDimensions(9);
+
+            if (_inputBoxes.Length > _scrollBar.LinesToRender)
+            {
+                for (int i = 0; i < _inputBoxes.Length; i++)
+                {
+                    var currBox = _inputBoxes[i];
+                    var currLabel = _inputLabels[i];
+                    if (i < _scrollBar.ScrollOffset)
+                    {
+                        currBox.Visible = false;
+                        currLabel.Visible = false;
+                        continue;
+                    }
+
+                    if (i < _scrollBar.LinesToRender + _scrollBar.ScrollOffset)
+                    {
+                        currBox.Visible = true;
+                        currLabel.Visible = true;
+
+                        int relativeIndex = i - _scrollBar.ScrollOffset;
+                        currBox.DrawPosition = new Vector2(currBox.DrawPosition.X, 69 + relativeIndex * 24);
+                        currLabel.DrawPosition = new Vector2(currLabel.DrawPosition.X, 69 + relativeIndex * 24);
+                    }
+                    else
+                    {
+                        currBox.Visible = false;
+                        currLabel.Visible = false;
+                    }
+                }
+            }
+            else if (_inputBoxes.Any(_item => _item.Visible) || _inputLabels.Any(_item => !_item.Visible))
```

Comment:
Since the above condition is always true, this `else if` branch can be removed entirely since it is never executed